### PR TITLE
Workaround for #1057776, install all SLES modules on JeOS

### PR DESCRIPTION
**Pull request by mnowaksuse**
With `ENABLE_ALL_SCC_MODULES=1`: http://assam.suse.cz/tests/682#step/suseconnect_scc/16
Without `ENABLE_ALL_SCC_MODULES=1`: http://assam.suse.cz/tests/683#step/suseconnect_scc/12

### CODE REVIEW

File: lib/apachetest.pm [hunk 1]
```diff
@@ -78,6 +78,10 @@ sub setup_apache2 {
 
     # Check server certificate is available, which should be generated during apache2-mod_nss installation
     if ($mode =~ m/NSS/) {
+        if (check_var('VERSION', 15)) {
```
Comment (okurz) [inline lib/apachetest.pm]: why not use `is_sle && sle_version_at_least('15')`? We don't want to check the version again for '15-SP1' or for Leap, '15.1'

File: tests/console/postgresql96server.pm [hunk 2]
```diff
@@ -27,7 +27,6 @@ sub run {
 
     # check the status
     assert_script_run 'systemctl show -p ActiveState postgresql.service | grep ActiveState=active';
-    assert_script_run 'systemctl show -p SubState postgresql.service | grep SubState=running';
```
Comment (okurz) [inline tests/console/postgresql96server.pm]: hm, I'm not convinced. I checked myself as described in https://bugzilla.suse.com/show_bug.cgi?id=1060639#c2 and the state is 'active (exited)' even though the postgres process is running. So maybe there *is* something wrong with the service. Maybe the first step would be to add the test module also to openSUSE Tumbleweed tests.

File: lib/apachetest.pm (hunk 1 above)
Comment (mnowaksuse) [reply lib/apachetest.pm]: I added SLE as well as Leap provision for versions 15+.

File: tests/console/postgresql96server.pm (hunk 2 above)
Comment (mnowaksuse) [reply tests/console/postgresql96server.pm]: Glad I did not convince you: https://bugzilla.suse.com/show_bug.cgi?id=1060639#c9 :).

Comment (mnowaksuse) [reply tests/console/postgresql96server.pm]: Commit dropped.

File: tests/jeos/sccreg.pm [hunk 3]
```diff
@@ -26,7 +26,7 @@ sub run {
     assert_script_run "SUSEConnect --url=$url -r $scccode";
     assert_script_run 'SUSEConnect --list-extensions';
     # Install Server-like modules
-    for my $module ('basesystem', 'scripting', 'legacy') {
+    for my $module ('basesystem', 'scripting', 'desktop-applications', 'server-applications', 'development-tools', 'legacy') {
```
Comment (okurz) [inline tests/jeos/sccreg.pm]: Maybe you can now use the same approach as in https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/3776/files . So could be something like:

```perl
    foreach ((split(',', $registration::SLE15_DEFAULT_MODULES{get_required_var('SLE_PRODUCT')}), qw(sdk legacy)) { …
```

but we can also leave it explicit for now. Your choice.

File: lib/apachetest.pm [hunk 4]
```diff
@@ -78,6 +78,11 @@ sub setup_apache2 {
 
     # Check server certificate is available, which should be generated during apache2-mod_nss installation
     if ($mode =~ m/NSS/) {
+        # Workaround for bsc#1057776 - Wrong permissions in /etc/apache2/mod_nss.d/
+        if (sle_version_at_least('15') || leap_version_at_least('15')) {
```
[the remark this answers is not in the document]
Comment (mnowaksuse) [reply lib/apachetest.pm]: Hopefully having it right this time. Thanks so far.

File: tests/jeos/sccreg.pm (hunk 3 above)
Comment (mnowaksuse) [reply tests/jeos/sccreg.pm]: I put in a variant on that in `console/suseconnnect_scc` test which I can now use use in JeOS. @yxususe let me know if the addition is fine with you.

**Comment by okurz** [comment]
LGTM

@yxususe what do you think?

**Comment by yxususe** [comment]
Great, thanks Michal.

**Comment by mnowaksuse** [comment]
Thanks @yxususe! Merging.

**Pull request merged**